test/src/XcooBee/Core/Api/UsersTest.php

```diff
@@ -87,8 +87,65 @@ public function testGetConversations_UseConfig()
             'apiSecret' => 'testapisecret' 
         ]);
     }
-    
-    public function testSendUserMessage() 
+
+    public function testGetUserPublicKey()
+    {
+        $usersMock = $this->_getMock(\XcooBee\Core\Api\Users::class, [
+            '_request' => true,
+        ]);
+
+        $usersMock->expects($this->once())
+                ->method('_request')
+                ->will($this->returnCallback(function ($query, $params) {
+                            $this->assertEquals(['xid' => 'testXid'], $params);
+                        }));
+
+        $usersMock->getUserPublicKey('testXid');
```

Comment:
in this case mock `_request` to return valid response with data and `pgp_public_key` and compare response of `getUserPublicKey` to your mocked key